IrrlichtMt: Implement mip-mapping for RTTs

SmallJoker opened the pull request:
This can be helpful to draw fonts memory-efficiently at varying scales.

I originally wrote this as an alternative approach to #16267. It turned out that the opengl and opengl3 drivers did not support mip-mapped RTT's (render target textures), which would be necessary for that. This PR implements that.

## To do

This PR is Ready for Review.

## How to test

1. Revert/undo the 2nd commit
2. Compile and join a world
3. Look around. The text on the bottom will scale based on the view direction
4. Repeat the same thing with the opengl/opengl3 driver respectively.
5. Use the opengl3 driver and [renderdoc](https://renderdoc.org/) to confirm that the generated texture is indeed mip-mapped

SmallJoker commented on irr/src/COpenGLCoreTexture.h:
```diff
@@ -400,6 +393,13 @@ class COpenGLCoreTexture : public ITexture
+			GL.Hint(GL_GENERATE_MIPMAP_HINT, GL_FASTEST);
+		else if (Driver->getTextureCreationFlag(ETCF_OPTIMIZED_FOR_QUALITY))
+			GL.Hint(GL_GENERATE_MIPMAP_HINT, GL_NICEST);
+		else
```
At this point I think it's simplest to update it once on init and then keep it throughout the entire session. This is too much of a niche use-case.

sfan5 replied:
> I agree that `ETCF_OPTIMIZED_FOR_*` could probably be removed, but I'd like to do that in another cleanup PR.

sure.

SmallJoker commented on irr/src/OpenGL/Driver.cpp:
```diff
@@ -1716,6 +1708,13 @@ bool COpenGL3DriverBase::setRenderTargetEx(IRenderTarget *target, u16 clearFlag,
+		auto textures = CurrentRenderTarget->getTexture();
+		for (size_t i = 0; i < textures.size(); ++i)
+			textures[i]->regenerateMipMapLevels();
+	}
```
It will only be run once if mip maps are enabled on the RTT. (reference to your RTT-specific flag below)

sfan5 replied:
so can it be called externally? regenerating mipmaps every time an RTT is used (`setRenderTargetEx`) sounds like a potential silent performance killer

SmallJoker replied:
`regenerateMipMapLevels` checks against `HasMipMaps` and skips the entire regenerate part if that is not enabled. Is this not sufficient?

SmallJoker commented on irr/src/CNullDriver.cpp:
```diff
@@ -84,13 +84,16 @@ CNullDriver::CNullDriver(io::IFileSystem *io, const core::dimension2d<u32> &scre
+	InitMaterial2D.UseMipMaps = true;
+	InitMaterial2D.forEachTexture([](video::SMaterialLayer &tex) {
+		// Best result on fonts
+		tex.MinFilter = video::ETMINF_LINEAR_MIPMAP_LINEAR;
```
Update: The aforementioned bug does not occur to me when `tex.LODBias = -8;` is specified.
Tested using [PRANG](https://content.luanti.org/packages/luk3yx/prang/), where the artefacts do neither appear with `ETMINF_LINEAR_MIPMAP_NEAREST` (reverted commit) nor `ETMINF_LINEAR_MIPMAP_LINEAR` (this PR). `opengl` and `opengl3` show no difference either.

@grorp Would you please be so nice to test those combinations to ensure this PR causes no such regression on your system?

SmallJoker commented:
About https://github.com/luanti-org/luanti/pull/16434#discussion_r2296558265
![comparison](https://github.com/user-attachments/assets/0e5d7124-5ba3-4b88-b12a-d850321e2e10)

Hence, I reverted these changes in f9afbfa and converted them to comments instead. Materials shall be used depending on the application.
